### DISCUSSION TITLE
How to search tags which are in the YAML front matter?

### DISCUSSION THREAD

**Discussion by serifold** [foambubble/foam#1285] [Q&A] [answered]
For ordinary tags, e.g. `#tag1`, I know they can be found by search the keyword  `#tag1`. However, if I have some tags in YAML front matter, e.g.

```yaml
---
tags: tag1 tag2
---
```

I can't find them by searching such as `#tag1`. If I change `tags: tag1 tag2` to `tags: #tag1 #tag2`, then Foam won't recognize them as tags. 

Is there something wrong the way I use tags in YAML front matter? If not, could we add a search button in the `Tag Explorer` bar, which can search all tags? Or just add a command such as `Foam: Search tags` that can treat YAML tags correctly.

**Comment by riccardoferretti** [comment]
By expanding the tag in the tag explorer you get a list of all the places where they are present, is that not what you are looking for?

**Reply by serifold** [reply]
Not really.

When tags are still a not large amount, it's OK to scan and find specific tags in `Tag Explorer`. However, as the amount of tags increasing, in my opinion, it's more efficient that if we can find tags by directly search them, e.g. I type `#tag1`, then all tags either in the `#tag1` form or in in YAML front matter `tags: ..., tag1, ...` can be found.

Considering `Ctrl + Shift + F` is mainly used for searching normal text, although it can search tags in `#tag1` form, but it can't search tags in YAML front matter correctly, as I said previously, maybe add a `search` button in `Tag Explorer` bar or just add a command like `Foam: Search tags` which can treat tags in YAML front matter correctly can make searching tags more efficient and convenient.

**Reply by riccardoferretti** [reply, marked as answer]
I understand. It's not on top of my priority list, but if someone wants to give it a stab I'd be happy to support.
Will track this with an issue

**Reply by serifold** [reply]
Thank you. I'll try to deal with it in my spare time to see if I can commit something.
